src/main/scala/mesosphere/marathon/core/task/state/MarathonTaskStatus.scala

```diff
@@ -0,0 +1,100 @@
+package mesosphere.marathon.core.task.state
+
+import org.apache.mesos
+
+// TODO ju handle garbage flag
+sealed trait MarathonTaskStatus {
+  def toMesosStateName: String = "TASK_" + toString.toUpperCase
+}
+
+object MarathonTaskStatus {
+  import org.apache.mesos.Protos.TaskState._
+
+  def fromMesosStateName(mesosStateName: String): Option[MarathonTaskStatus] = all.collectFirst {
+    case status: MarathonTaskStatus if status.toMesosStateName == mesosStateName => status
+  }
+
+  // If we're disconnected at the time of a TASK_LOST event, we will only get the update during
+  // a reconciliation. In that case, the specific reason will be shadowed by REASON_RECONCILIATION.
+  // Since we don't know the original reason, we need to assume that the task might come back.
+  val MightComeBack: Set[mesos.Protos.TaskStatus.Reason] = Set(
+    mesos.Protos.TaskStatus.Reason.REASON_RECONCILIATION,
+    mesos.Protos.TaskStatus.Reason.REASON_SLAVE_DISCONNECTED,
+    mesos.Protos.TaskStatus.Reason.REASON_SLAVE_REMOVED
+  )
+
+  val WontComeBack: Set[mesos.Protos.TaskStatus.Reason] = {
+    mesos.Protos.TaskStatus.Reason.values().toSet.diff(MightComeBack)
+  }
+
+  trait Terminal
+
+  //scalastyle:off cyclomatic.complexity
+  def apply(taskStatus: mesos.Protos.TaskStatus): MarathonTaskStatus = {
+    taskStatus.getState match {
+      case TASK_ERROR => Error
+      case TASK_FAILED => Failed
+      case TASK_FINISHED => Finished
+      case TASK_KILLED => Killed
+      case TASK_KILLING => Killing
+      case TASK_LOST => taskStatus.getReason match {
+        case state: mesos.Protos.TaskStatus.Reason if WontComeBack(state) => Gone
+        case state: mesos.Protos.TaskStatus.Reason if MightComeBack(taskStatus.getReason) => Unreachable
+        case _ => Lost
+      }
+      case TASK_RUNNING => Running
+      case TASK_STAGING => Staging
+      case TASK_STARTING => Starting
+    }
+  }
+
+  /**
+    * (Temporarily) mapping of mesos.Protos.TaskStatus.TASK_LOST to MarathonTaskStatus.
+    * Should be removed as soon as `Gone`, `Unreachable` and `Lost` are native Mesos task states.
+    *
+    * @return list of possible mappings of the mesos.Protos.TaskStatus.TASK_LOST
+    */
+  lazy val mightBeLost: Set[MarathonTaskStatus] = Set(Gone, Unreachable, Lost)
+
+  lazy val all: Set[MarathonTaskStatus] = Set(Reserved, Created, Error, Failed, Finished, Killed, Killing, Lost, //
+    Running, Staging, Starting, Unreachable, Gone, Unknown)
+
+  // Marathon specific states
+  // RESERVED
```

Comment:
I am not sure if these capitalized comments add information or disturb readability. If you can add a doc line describing what the state _means_, this would be far better imo.